bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/converter/GenericItemConverter.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.http.internal.converter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.http.internal.config.HttpChannelConfig;
+import org.openhab.binding.http.internal.transform.ValueTransformation;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * The {@link GenericItemConverter} implements simple conversions for different item types
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class GenericItemConverter<T extends State> extends AbstractTransformingItemConverter {
+    private final Class<T> clazz;
+
+    public GenericItemConverter(Class<T> clazz, Consumer<State> updateState, Consumer<Command> postCommand,
+            @Nullable Consumer<String> sendHttpValue, ValueTransformation stateTransformations,
+            ValueTransformation commandTransformations, HttpChannelConfig channelConfig) {
+        super(updateState, postCommand, sendHttpValue, stateTransformations, commandTransformations, channelConfig);
+        this.clazz = clazz;
+    }
+
+    protected State toState(String value) {
+        try {
+            return clazz.getConstructor(String.class).newInstance(value);
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException
+                | IllegalAccessException e) {
+            return UnDefType.NULL;
+        }
+    }
```

Comment:
I don't know if generics would complicate things. But if the compiler will accept `StringType::new` as a `Function<String, State>` instance then it should be possible to replace the following without difficulty.

```suggestion
public class GenericItemConverter extends AbstractTransformingItemConverter {
    private final Function<String, State> toStateFunction;

    public GenericItemConverter(Function<String, State> toStateFunction, Consumer<State> updateState, Consumer<Command> postCommand,
            @Nullable Consumer<String> sendHttpValue, ValueTransformation stateTransformations,
            ValueTransformation commandTransformations, HttpChannelConfig channelConfig) {
        super(updateState, postCommand, sendHttpValue, stateTransformations, commandTransformations, channelConfig);
        this.toStateFunction = toStateFunction;
    }

    protected State toState(String value) {
        return toStateFunction.apply(value);
    }
```